clustering/infinispan/extension/src/test/java/org/jboss/as/clustering/infinispan/subsystem/TransformersTestCase.java

```diff
@@ -184,6 +211,126 @@ private void testTransformation(InfinispanModel model, ModelTestControllerVersio
             if (transaction.hasDefined(TransactionResourceDefinition.Attribute.MODE.getDefinition().getName())) {
                 Assert.assertEquals(TransactionMode.NONE.name(), transaction.get(TransactionResourceDefinition.Attribute.MODE.getDefinition().getName()).asString());
             }
+
+            // Test properties operations
+            propertiesMapOperationsTest(services, version);
+        }
+    }
+
+    private void propertiesMapOperationsTest(KernelServices services, ModelVersion version) throws Exception {
+        final String cacheContainer = "maximal";
+
+        final String testProperty1 = "testProperty1";
+        final String testProperty2 = "testProperty2";
+        final String testProperty3 = "testProperty3";
+        final String testProperty4 = "testProperty4";
+
+        final List<PathAddress> cacheStoreAddresses = new LinkedList<>();
+
+        // Current addresses
+        cacheStoreAddresses.add(getBinaryKeyedJDBCCacheStoreAddress(cacheContainer, ReplicatedCacheResourceDefinition.WILDCARD_PATH.getKey(), "cache-with-binary-keyed-store"));
+        cacheStoreAddresses.add(getCustomCacheStoreAddress(cacheContainer, ReplicatedCacheResourceDefinition.WILDCARD_PATH.getKey(), "repl"));
+        cacheStoreAddresses.add(getFileCacheStoreAddress(cacheContainer, LocalCacheResourceDefinition.WILDCARD_PATH.getKey(), "local"));
+        cacheStoreAddresses.add(getMixedKeyedJDBCCacheStoreAddress(cacheContainer, DistributedCacheResourceDefinition.WILDCARD_PATH.getKey(), "dist"));
+        cacheStoreAddresses.add(getRemoteCacheStoreAddress(cacheContainer, InvalidationCacheResourceDefinition.WILDCARD_PATH.getKey(), "invalid"));
+        cacheStoreAddresses.add(getStringKeyedJDBCCacheStoreAddress(cacheContainer, ReplicatedCacheResourceDefinition.WILDCARD_PATH.getKey(), "cache-with-string-keyed-store"));
+
+        // Legacy addresses
+        // TODO transformations do not kick in for legacy addresses
+//        cacheStoreAddresses.add(getBinaryKeyedJDBCCacheStoreLegacyAddress(cacheContainer, ReplicatedCacheResourceDefinition.WILDCARD_PATH.getKey(), "cache-with-binary-keyed-store"));
+//        cacheStoreAddresses.add(getCustomCacheStoreLegacyAddress(cacheContainer, ReplicatedCacheResourceDefinition.WILDCARD_PATH.getKey(), "repl"));
+//        cacheStoreAddresses.add(getFileCacheStoreLegacyAddress(cacheContainer, LocalCacheResourceDefinition.WILDCARD_PATH.getKey(), "local"));
+//        cacheStoreAddresses.add(getMixedKeyedJDBCCacheStoreLegacyAddress(cacheContainer, DistributedCacheResourceDefinition.WILDCARD_PATH.getKey(), "dist"));
+//        cacheStoreAddresses.add(getRemoteCacheStoreLegacyAddress(cacheContainer, InvalidationCacheResourceDefinition.WILDCARD_PATH.getKey(), "invalid"));
+//        cacheStoreAddresses.add(getStringKeyedJDBCCacheStoreLegacyAddress(cacheContainer, ReplicatedCacheResourceDefinition.WILDCARD_PATH.getKey(), "cache-with-string-keyed-store"));
```

Comment:
With a bit of commenting/uncommenting I tried these two addresses:
    Current: /subsystem=infinispan/cache-container=maximal/replicated-cache=cache-with-binary-keyed-store/store=binary-jdbc
    Old/aliased address /subsystem=infinispan/cache-container=maximal/replicated-cache=cache-with-binary-keyed-store/binary-keyed-jdbc-store= BINARY_KEYED_JDBC_STORE

Debugging for the two cases when trying to find the transformers in this following stack:

    resolveOperationTransformer():291, OperationTransformerRegistry {org.jboss.as.controller.registry}
    resolveOperationTransformer():299, OperationTransformerRegistry {org.jboss.as.controller.registry}
    resolveOperationTransformer():299, OperationTransformerRegistry {org.jboss.as.controller.registry}
    resolveOperationTransformer():299, OperationTransformerRegistry {org.jboss.as.controller.registry}
    resolveOperationTransformer():109, OperationTransformerRegistry {org.jboss.as.controller.registry}
    resolveTransformer():152, TransformationTargetImpl {org.jboss.as.controller.transform}
    transformOperation():61, TransformersImpl {org.jboss.as.controller.transform}
    transformOperation():112, MainKernelServicesImpl {org.jboss.as.subsystem.test}
    executeInMainAndGetTheTransformedOperation():200, MainKernelServicesImpl {org.jboss.as.subsystem.test}
    executeOpInBothControllersWithAttachments():126, PropertiesTestUtil {org.jboss.as.clustering.controller}
    propertiesMapOperationsTest():251, TransformersTestCase {org.jboss.as.clustering.infinispan.subsystem}

When it reaches the point where it is looking for the children of /subsystem=infinispan/cache-container=maximal/replicated-cache=cache-with-binary-keyed-store, it appears that there is a subregistry for 'store' which in turn contains the transformers for 'binary-jdbc'. However, there is no subregistry for 'binary-keyed-jdbc-store', and so no transformer for BINARY_KEYED_JDBC_STORE.

So whatever registers the transformers for the children of /subsystem=infinispan/cache-container=maximal/replicated-cache=cache-with-binary-keyed-store also needs to register the child transformation for binary-keyed-jdbc-store= BINARY_KEYED_JDBC_STORE